docs/source/tutorials/creating_meshtype/creating_3d_tube.rst

```diff
@@ -0,0 +1,236 @@
+                            coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D2_DS1DS3, 1, zero)
+                            coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D2_DS2DS3, 1, zero)
+                            coordinates.setNodeParameters(cache, -1, Node.VALUE_LABEL_D3_DS1DS2DS3, 1, zero)
+                        nodeIdentifier += 1
+
+Here, we define that :math:`\zeta_1, \zeta_2, \zeta_3` are the element parameters along the tube, through the wall, and around the tube respectively. We can find the derivatives by multiplying the arc-length
+based derivatives and the corresponding scale factor. In this case, we set all cross derivatives to zero.
+
+With all nodal values set, we need to construct elements. Unlike the 3D cube example, where we only
+have 1 element, arranging node identifiers in this case can be a little bit tricky.
+
+.. code-block:: python
+
+    ...
+    class MeshType_3d_tube3(Scaffold_base):
+        ...
+        def generateBaseMesh(cls, region, options):
+            ...
+            elementIdentifier = 1
+            multiplier = (elementsCountThroughWall + 1) * (elementsCountAlong + 1)
+            totalNodes = (elementsCountAlong + 1) * (elementsCountThroughWall + 1) * elementsCountAround
+
+            for e3 in range(elementsCountAround):
+                for e2 in range(elementsCountThroughWall) :
+                    for e1 in range(elementsCountAlong):
+                        element = mesh.createElement(elementIdentifier, elementtemplate)
+                        n1 = 1 + e1 + (elementsCountAlong + 1) * e2 + e3 * multiplier
+                        n2 = n1 + 1
+                        n3, n4 = n1 + elementsCountAlong + 1 , n2 + elementsCountAlong + 1
+                        n5, n6 = n1 + multiplier, n2 + multiplier
+                        n7, n8 = n5 + elementsCountAlong + 1, n6 + elementsCountAlong + 1
+                        nodeIdentifiers = [*map(lambda value: value - totalNodes if value > totalNodes else value,
+                                              [n1, n2, n3, n4, n5, n6, n7, n8])]
+                        result = element.setNodesByIdentifier(eft, nodeIdentifiers)
+                        print(elementIdentifier,nodeIdentifiers)
+                        elementIdentifier += 1
+            fm.endChange()
+            return []
+
+If you look back, our node identifiers accumulate in the following order:
+
+* along the tube :pyth:`e1`
+* through the wall :pyth:`e2`
+* around the tube :pyth:`e3`
+
+Thus, we can address the first node identifier in each element, then use the first node identifier to locate others. In the
+last iteration of :pyth:`e3`, nodes should return to the starting position. Thus I subtracted :pyth:`totalNodes`
+from each node identifier if it exceeds the total number of nodes, to complete the loop. If you are struggling
+to picture this, check the figure below for better understanding.
```

Comment:
check the figure below for better understanding ->see the figure below for more information.